google_geocoder.go

```diff
@@ -116,29 +137,68 @@ func (g *GoogleGeocoder) Geocode(address string) (*Point, error) {
 	return point, nil
 }
 
-func googleGeocodeQueryStr(address string) (string, error) {
-	url_safe_query := url.QueryEscape(address)
-
-	var queryStr = bytes.NewBufferString("")
-	_, err := queryStr.WriteString(fmt.Sprintf("address=%s", url_safe_query))
-	if err != nil {
-		return "", err
-	}
+func googleFormattedRequestStr(params string) (string, error) {
+	queryStr := fmt.Sprintf("%s&%s", "sensor=false", params)
 
 	if GoogleAPIKey != "" {
-		_, err := queryStr.WriteString(fmt.Sprintf("&key=%s", GoogleAPIKey))
+		queryStrBuffer := bytes.NewBufferString(queryStr)
```

Comment:
Would it make sense here to have a set of authentication schemas that you can select when you make the Geocoder?

For example:

```go
type GoogleAuthSchema int

const(
  GoogleClientSideAuth = iota
  GoogleForWorkAuth
)

type GoogleGeocoder struct {
    HttpClient *http.Client
    AuthSchema GoogleAuthSchema
}

// sometime later...

g := &GoogleGeocoder{
    HttpClient: &http.Client{},
    AuthSchema: GoogleForWorkAuth,
}
```

That way, there's less of a chance someone could step on their own toes by accidentally setting both an API Key and a set of "For Work" auth credentials.